docs/examples/machine_translation/transformer.md

```diff
@@ -49,79 +47,75 @@ wmt_transformer_model, wmt_src_vocab, wmt_tgt_vocab = \
+We then load the newstest2014 segment in WMT 2014 English-German test dataset for evaluation purpose.
 
 The following shows how to process the dataset and cache the processed dataset
 for the future use. The processing steps include:
 
-* 1) clip the source and target sequences
-* 2) split the string input to a list of tokens
-* 3) map the string token into its index in the vocabulary
-* 4) append EOS token to source sentence and add BOS and EOS tokens to target sentence.
+1) clip the source and target sequences
+2) split the string input to a list of tokens
+3) map the string token into its index in the vocabulary
+4) append EOS token to source sentence and add BOS and EOS tokens to target sentence.
 
-Let's first look at the WMT 2014 corpus.
+Let's first look at the WMT 2014 corpus. GluonNLP provides [WMT2014BPE](../../api/data.rst#gluonnlp.data.WMT2014BPE)
+and [WMT2014](../../api/data.rst#gluonnlp.data.WMT2014) classes. The former contains BPE-tokenized dataset, while
+the later contains the raw text. Here, we use the former for scoring, and the later for
+demonstrating actual translation.
 
 ```{.python .input}
 import hyperparameters as hparams
 
 wmt_data_test = nlp.data.WMT2014BPE('newstest2014',
                                     src_lang=hparams.src_lang,
-                                    tgt_lang=hparams.tgt_lang,
-                                    full=False)
+                                    tgt_lang=hparams.tgt_lang)
 print('Source language %s, Target language %s' % (hparams.src_lang, hparams.tgt_lang))
+print('Sample: "{}"'.format(wmt_data_test[0]))
 
-wmt_data_test[0]
-```
-
-```{.python .input}
 wmt_test_text = nlp.data.WMT2014('newstest2014',
                                  src_lang=hparams.src_lang,
-                                 tgt_lang=hparams.tgt_lang,
-                                 full=False)
-wmt_test_text[0]
-```
+                                 tgt_lang=hparams.tgt_lang)
+print('Sample text: "{}"'.format(wmt_test_text[0]))
 
-We then generate the target gold translations.
-
-```{.python .input}
-wmt_test_tgt_sentences = list(wmt_test_text.transform(lambda src, tgt: tgt))
-wmt_test_tgt_sentences[0]
+wmt_test_tgt_sentences = wmt_test_text.transform(lambda src, tgt: tgt)
+print('Sample target sentence: "{}"'.format(wmt_test_tgt_sentences[0]))
 ```
 
 ```{.python .input}
 import dataprocessor
 
 print(dataprocessor.TrainValDataTransform.__doc__)
-```
 
-```{.python .input}
-wmt_transform_fn = dataprocessor.TrainValDataTransform(wmt_src_vocab, wmt_tgt_vocab, -1, -1)
+# wmt_transform_fn includes the four preprocessing steps mentioned above.
+wmt_transform_fn = dataprocessor.TrainValDataTransform(wmt_src_vocab, wmt_tgt_vocab)
 wmt_dataset_processed = wmt_data_test.transform(wmt_transform_fn, lazy=False)
 print(*wmt_dataset_processed[0], sep='\n')
-```
 
-### Create Sampler and DataLoader for WMT 2014 Dataset
+def get_length_index_fn():
```

Comment:
We might want to hide this, we can integrate this `def get_length_index_fn()` in `dataprocessor`.